### PR TITLE
Update code to zope for py3 and not give errors on python3 compileall

### PR DESCRIPTION
**Pull request by adiroiban**
Scope
======

This is part of fixing #55

It builds on top of #83 and it does the minimum changes required to have the code compile on py3

Changes
=======

Do minimal changes to pass compileall

Update zope interface usage.

I have enabled the python3 tests to see what fails in preparation for future PRs.

As a drive-by I have re-arrange the order of some imports to have stdlib first, then other libs and last our own imports.

As a drive-by I fixed the indention of travis.yml to 2 spaces... it was a mix of 4 and 2 spaces.

How to check
=========

no errors should be present when compiling with python3

```
$ python3 -m compileall ldaptor
```

### CODE REVIEW

**Comment by psi29a** [comment]
Also LGTM, changelog entry needed to mark the updates to libraries please. :)

**Comment by adiroiban** [comment]
Thanks for your review. That was fast :)

@psi29a ready for a new review as I have retargeted this to master... there were some unnecessary conflicts due to squash merging.

Let  me know if the level and wording for the relese notes are ok.

---

I will update the docs about contributing in a new branch where I also want to create a template for the PR.

File: ldaptor/samba/smbpassword.py [hunk 1]
```diff
@@ -1,9 +1,15 @@
-import string, warnings
+import warnings
+try:
+    import string
+    maketrans = string.maketrans
+except ImportError:
```
Comment (adiroiban) [inline ldaptor/samba/smbpassword.py]: While this does not gives errors, this is broken as here we should catch AttributeError

I am fixing it a branch which is a child of a child of this one so to simplify the merge  I think that is ok to merge it as it is... as with the squash merge things will get messy

File: ldaptor/test/test_attributeset.py [hunk 2]
```diff
@@ -43,26 +42,31 @@ class TestSetOperations(unittest.TestCase):
     def testDifference(self):
         a = attributeset.LDAPAttributeSet('k', ['b', 'c', 'd'])
         b = attributeset.LDAPAttributeSet('k', ['b', 'c', 'e'])
-        self.assertEqual(a - b, sets.Set(['d']))
+        self.assertEqual(a - b, {'d'})
```
Comment (s0undt3ch) [inline ldaptor/test/test_attributeset.py]: If ldaptor still supports 2.6, `set`s cannot be defined like this.
It's probably a good time to drop 2.6 support while gaining 3.x support?

File: docs/source/NEWS.rst [hunk 3]
```diff
@@ -14,6 +14,8 @@ Changes
 ^^^^^^^
 
 - Using modern classmethod decorator instead of old-style method call.
+- Usage of zope.interfaces was updated in preparation for python3 port.
+- Code was updates to pass `python3 -m compileall` in preparation for py3 port.
```
[the remark this answers is not in the document]
Comment (adiroiban) [reply docs/source/NEWS.rst]: thanks

File: ldaptor/test/test_attributeset.py (hunk 2 above)
Comment (adiroiban) [reply ldaptor/test/test_attributeset.py]: Thanks for the review.

I don't think that 2.6 is supported... and is hard to support both 2.6 and python 3

And 2.6 is no longer supported by upstream Python for a very long time

I see that tests are executed only under 2.7 https://github.com/twisted/ldaptor/commits/master/.travis.yml

I think 2.6 was already silently dropped :)

I see that in 14.0.0 it states that 2.6 is supported but it was removed without a changelog by @psi29a here https://github.com/twisted/ldaptor/commit/b6b4ea6fc55597977dbf5a64a6c8dca44eb12fe3#diff-354f30a63fb0907d4ad57269548329e3 more than 1 year ago

Comment (s0undt3ch) [reply ldaptor/test/test_attributeset.py]: Then, awesome work!

Comment (psi29a) [reply ldaptor/test/test_attributeset.py]: Twisted dropped 2.6 support, so we should follow their example.

**Comment by psi29a** [comment]
Before I merge this in, there is a conflict to be resolved here:
.travis.yml

**Comment by adiroiban** [comment]
I have fixed the merge in Travis. I can also merge it.

If we only want squash commit in merge, I think that we should disable all the other modes.

I have updated the settings of the repo. I have also removed wiki and projects as they are not used.  Let me know if you want the old settings back.

**Comment by psi29a** [comment]
@adiroiban  Sure, I don't believe a wiki and the rest as necessary to this project since we have our own RTD.

**Comment by psi29a** [comment]
We can just do merge, instead of squash and merge if that creates too many issues.  Just a fall-over from how I handle other projects where people will bring in a PR with many many commits which are just small fixes/typos.

**Comment by adiroiban** [comment]
@psi29a I am ok with merge and squash. It will get extra work at merge time, but I prefer to have a clean history without "Fix typo." or "I am stupid" commit messages :) 

I merge it one but read the history many times :)

so +1 from me for the merge with squash.

**Pull request merged**